Nio/Authentication/LoginView.swift

```diff
@@ -36,6 +37,23 @@ struct LoginContainerView: View {
 
         store.login(username: username, password: password, homeserver: homeserverURL)
     }
+    
+    private func guessHomeserverURL() {
+        if !username.isEmpty && homeserver.isEmpty {
+            let userparts = username.split(separator: ":")
+            if userparts.count < 2 {
+                return
+            }
```

Comment:
I'd suggest using `guard` instead, it reads a bit easier regarding the check. I also went with `== 2`, since everything else would result in issues as well.
```suggestion
            guard userparts.count == 2 else { return }
```